sdcm/nemesis.py

```diff
@@ -274,6 +275,46 @@ def disrupt_nodetool_refresh(self, big_sstable=True, skip_download=False):
             cmd = "select * from keyspace1.standard1 where key=0x314e344b4d504d4b4b30"
             node.remoter.run('cqlsh -e "{}" {}'.format(cmd, node.private_ip_address), verbose=True)
 
+    def disrupt_nodetool_enospc(self, sleep_time=30, all_nodes=False):
+        if all_nodes:
+            nodes = self.cluster.nodes
+        else:
+            nodes = [self.target_node]
+        self._set_current_disruption('Enospc test on {}'.format(nodes))
+        for node in nodes:
+            result = node.remoter.run('cat /proc/mounts')
+            if '/var/lib/scylla' not in result.stdout:
+                self.log.error("Scylla doesn't use an individual storage, skip enospc test")
+                continue
+
+            # get the size of free space (default unit: KB)
+            result = node.remoter.run("df -l|grep '/var/lib/scylla'")
+            free_space_size = result.stdout.split()[3]
+
+            occupy_space_size = int(free_space_size) * 99 / 100
+            occupy_space_cmd = 'sudo fallocate -l {}K /var/lib/scylla/occupy_99percent'.format(occupy_space_size)
+            self.log.debug('Cost 99% free space on /var/lib/scylla/ by {}'.format(occupy_space_cmd))
+
+            # check original ENOSPC error
+            orig_errors = node.search_database_log('No space left on device')
+            result = node.remoter.run(occupy_space_cmd, ignore_status=True, verbose=True)
+            wait.wait_for(func=lambda: len(node.search_database_log('No space left on device')) > len(orig_errors),
+                          step=5,
+                          text='Wait for new ENOSPC error occurs in database')
```

Comment:
@amoskong , please add a timeout to to this wait, to make sure the test will proceed in case it couldn't reach /catch the ENOSPC.